internal/dag/gatewayapi_processor.go

```diff
@@ -106,9 +143,42 @@ func (p *GatewayAPIProcessor) Run(dag *DAG, source *KubernetesCache) {
 
 		// Process all the routes that match this Gateway.
 		for _, matchingRoute := range matchingRoutes {
-			p.computeHTTPRoute(matchingRoute)
+			hosts := p.computeHosts(matchingRoute)
+			p.computeHTTPRoute(matchingRoute, hosts, listenerSecret)
+		}
+	}
+}
+
+func (p *GatewayAPIProcessor) computeHosts(route *gatewayapi_v1alpha1.HTTPRoute) []string {
+	// Determine the hosts on the route, if no hosts
+	// are defined, then set to "*".
+	var hosts []string
+	if len(route.Spec.Hostnames) == 0 {
+		hosts = append(hosts, "*")
+	} else {
+		for _, host := range route.Spec.Hostnames {
+
+			hostname := string(host)
+			if isIP := net.ParseIP(hostname) != nil; isIP {
+				p.Errorf("hostname %q must be a DNS name, not an IP address", hostname)
```

Comment:
I assume these should eventually become conditions, any reason not to do that now?

Reply:
I'll update, the conditions merged before this so just got lost.